sdk/storage/inc/datalake/path_client.hpp

```diff
@@ -0,0 +1,249 @@
+     */
+    explicit PathClient(
+        const std::string& pathUri,
+        const PathClientOptions& options = PathClientOptions());
+
+    /**
+     * @brief Uploads data to be appended to a file. Data can only be appended to a file.
+     * @param data The data to be appended.
+     * @param offset This parameter allows the caller to upload data in parallel and control
+     *                 the order in which it is appended to the file.
+     *                 The value must be the offset where the data is to be appended.
+     *                 Uploaded data is not immediately flushed, or written, to the file. To flush,
+     *                 the previously uploaded data must be contiguous, the offset parameter must
+     *                 be specified and equal to the length of the file after all data has been
+     *                 written, and there must not be a request entity body included with the
+     *                 request.
+     * @param options Optional parameters to append data to the resource the path points to.
+     * @return PathAppendDataResponse
+     */
+    PathAppendDataResponse AppendData(
+        Azure::Core::Http::BodyStream* stream,
+        int64_t offset,
+        const PathAppendDataOptions& options = PathAppendDataOptions()) const;
+
+    /**
+     * @brief Flushes previous uploaded data to a file.
+     * @param offset This parameter allows the caller to upload data in parallel and control
+     *                 the order in which it is appended to the file.
+     *                 The value must be the offset where the data is to be appended.
+     *                 Uploaded data is not immediately flushed, or written, to the file. To flush,
+     *                 the previously uploaded data must be contiguous, the offset parameter must
+     *                 be specified and equal to the length of the file after all data has been
+     *                 written, and there must not be a request entity body included with the
+     *                 request.
+     * @param options Optional parameters to flush data to the resource the path points to.
+     * @return PathFlushDataResponse
+     */
+    PathFlushDataResponse FlushData(
+        int64_t offset,
+        const PathFlushDataOptions& options = PathFlushDataOptions()) const;
+
+    /**
+     * @brief Sets the owner, group, permissions, or access control list for a file or directory.
+     *        Note that Hierarchical Namespace must be enabled for the account in order to use
+     *        access control. Also note that the Access Control List (ACL) includes permissions for
+     *        the owner, owning group, and others, so the x-ms-permissions and x-ms-acl request
+     *        headers are mutually exclusive.
+     * @param options Optional parameters to set an access control to the resource the path points
+     *                to.
+     * @return PathFlushDataResponse
+     */
+    PathSetAccessControlResponse SetAccessControl(
```

Comment:
won't it be a little weird that a setter doesn't have a mandatory option? People may wonder, what would happen if I call this function without any parameter? I know you'd say it's optional in REST API. But that's a four-in-one REST API, it has other functionalities other than setting acl. In addition in both js and .net SDK, this function has a mandatory option.